File: Heat/ViewModels/Plants/CreatePlantViewModel.vb
Imports System.ComponentModel.DataAnnotations

Namespace ViewModels.Plants
    Public Class CreatePlantViewModel

        <Display(name:="Codice impianto")> _
        Property Code As Integer

        <Required> _
        <Display(name:="Nominativo")> _
        Property Name As String

        <Required> _
        <Display(name:="Indirizzo")> _
        Property Address As String

        <Display(name:="Numero Civico")> _
        Property StreetNumber As String

        <Display(name:="Palazzo")> _
        Property Building As String

        <Display(name:="Scala")> _
        Property Stair As String

        <Display(name:="Interno")> _
        Property Apartment As String

        <Required> _
        <Display(name:="Località")> _
        Property City As String

        <Display(name:="CAP")> _
        Property PostalCode As String

        <Display(name:="Area")> _
        Property Area As String

        <Display(name:="Zona")> _
        Property Zone As String

        <Display(name:="Provincia")> _
        Property District As String

        <Display(name:="Singola unità abitativa")> _
        Property IsSingleUnit As Boolean

        <Display(name:="Categoria energetica")> _
        Property EnergyCategory As EnergyCategoryEnum

        <Display(name:="Volume lordo riscaldato (m³)")> _
        Property GrossHeatedVolumeM3 As Single

        <Display(name:="Volume lordo raffrescato (m³)")> _
        Property GrossCooledVolumeM3 As Single


    End Class
End Namespace

File: Heat.ConvertedToC#/ViewModels/Plants/CreatePlantViewModel.cs
using System.ComponentModel.DataAnnotations;

namespace Heat.ViewModels.Plants
{
    public class CreatePlantViewModel
	{

		[Display(Name = "Codice impianto")]
		public int Code { get; set; }

		[Required()]
		[Display(Name = "Nominativo")]
		public string Name { get; set; }

		[Required()]
		[Display(Name = "Indirizzo")]
		public string Address { get; set; }

		[Display(Name = "Numero Civico")]
		public string StreetNumber { get; set; }

		[Display(Name = "Palazzo")]
		public string Building { get; set; }

		[Display(Name = "Scala")]
		public string Stair { get; set; }

		[Display(Name = "Interno")]
		public string Apartment { get; set; }

		[Required()]
		[Display(Name = "Località")]
		public string City { get; set; }

		[Display(Name = "CAP")]
		public string PostalCode { get; set; }

		[Display(Name = "Area")]
		public string Area { get; set; }

		[Display(Name = "Zona")]
		public string Zone { get; set; }

		[Display(Name = "Provincia")]
		public string District { get; set; }

		[Display(Name = "Singola unità abitativa")]
		public bool IsSingleUnit { get; set; }

		[Display(Name = "Categoria energetica")]
		public EnergyCategoryEnum EnergyCategory { get; set; }

		[Display(Name = "Volume lordo riscaldato (m³)")]
		public float GrossHeatedVolumeM3 { get; set; }

		[Display(Name = "Volume lordo raffrescato (m³)")]
		public float GrossCooledVolumeM3 { get; set; }


	}
}

